src/topology/generic_pipeline.rs

```diff
@@ -175,6 +175,12 @@ impl BatchSizer for [ResourceSpans] {
     }
 }
 
+impl BatchSizer for Vec<ResourceSpans> {
```

Comment:
I thought the array implementations above should be able to work for Vecs too. Maybe if the trait bounds above for `handle` were `[T]` instead of `Vec<T>` it would use the [] impl for BatchSizer?

Reply:
Done, you also need to change to as_slice `BatchSizer::size_of(otlp_payload.as_slice());`. 

Fine with this as have read it's best practice to pass accept slice rather than a specific container if you're not growing or taking ownership, this makes sense especially when exposing an API and I think it's actually more efficient? However, I'm not entirely sold on it for a `From` impl as we're not exposing a function call and we're only ever dealing with Vec of these types as that is what is required by the PB def. We could make add additional from OTLPFrom/Into's for [T] but again that seems like additional work as those are for ExportXXXServiceRequests, which will need to be converted back to Vec